### DISCUSSION TITLE
How to login via sms code?

### DISCUSSION THREAD

**Discussion by wangxingge** [tulir/whatsmeow#425] [WhatsApp protocol Q&A]
I want to login whatsapp via sms code validation. is there any avialable function can be used?

such as following:
timeoutChan := client.sendLoginSmsRequest()
go func(){for code:=range timeoutChan{
     client.validateSmsCode(code)
}}
timeoutChan<- "my sms code"

**Comment by Manjit2003** [comment]
It's not implemented in whatsmeow yet. But you can use the new pair code method with the pair-device branch. Just check the first few commits.

**Comment by jeffersonsc** [comment]
I found this code on this link "PairPhone" that receives a telephone number and shows a notification. My question is, could this be a new type of connection?

func (cli *Client) PairPhone(phone string, showPushNotification bool) (string, error) {

**Comment by hrizal** [comment]
i am not received any event when phone is connected with pair code.
how is your experience ?

**Reply by RezaOptic** [reply]
me too, I tried to connect with PairPhone but nothing happens until I get time-out error.

**Reply by jeffersonsc** [reply]
The code is present:

`// PairPhone generates a pairing code that can be used to link to a phone without scanning a QR code.

// The exact expiry of pairing codes is unknown, but QR codes are always generated and the login websocket is closed
// after the QR codes run out, which means there's a 160-second time limit. It is recommended to generate the pairing
// code immediately after connecting to the websocket to have the maximum time.

// The clientType parameter must be one of the PairClient* constants, but which one doesn't matter.
// The client display name must be formatted as `Browser (OS)`, and only common browsers/OSes are allowed
// (the server will validate it and return 400 if it's wrong).

// See https://faq.whatsapp.com/1324084875126592 for more info`

**Reply by RezaOptic** [reply]
I did it but the code stuck in PairPhone until it gave me a time-out error.

```
        err := client.Connect()
	if err != nil {
		panic(err)
	}
	s, err := client.PairPhone("00************", true, whatsmeow.PairClientChrome, "Chrome (Linux)")
	if err != nil {
		panic(err)
	}
```

**Reply by jeffersonsc** [reply]
Could you test by setting the parameter below as "Chrome (Linux)" in the connection name?

```go
waStore.DeviceProps.Os = proto.String("Chrome (Linux)")
```

**Reply by jeffersonsc** [reply]
I haven't had time to test that part of the system yet. Typically, I log in using the QR code image. As soon as I can test it, I'll come back and provide feedback.

**Reply by RezaOptic** [reply]
Sorry, the problem was mine. I shouldn't have used 00 at the beginning of my phone number.

**Comment by ghost** [comment]
How do I do this

**Comment by z5techbd** [comment]
Did you found any solution to do that? i am also searching for it. i want to login whatsapp with otp as a primary device.
